src/component/autocomplete.js

```diff
@@ -0,0 +1,312 @@
+
+    this.registerKeyDown(this.inputElement, 'down', event => {
+      this.selected = this.nextFoundResult(this.selected);
+      // do not move the cursor
+      event.preventDefault();
+    });
+
+    this.registerKeyDown(this.inputElement, 'up', event => {
+      this.selected = this.nextFoundResult(this.selected, true);
+      // do not move the cursor
+      event.preventDefault();
+    });
+
+    this.registerKeyDown(this.inputElement, 'enter', () => this.onSelect());
+
+    // tab closes the dropdown and jumps to the next tabable element (default behavior)
+    this.registerKeyDown(this.inputElement, 'tab', () => this.onSelect());
+  }
+
+  /**
+   * @param {Object} current selected item
+   * @param {Boolean} reversed when true gets the previous instead
+   *
+   * @returns {Object} the next of previous item
+   */
+  nextFoundResult(current, reversed) {
+    let index = (this.results.indexOf(current) + (reversed ? -1 : 1)) % (this.results.length);
+
+    if (index < 0) {
+      index = this.results.length - 1;
+    }
+
+    return this.results[index];
+  }
+
+  /**
+   * Set the text in the input to that of the selected item and set the
+   * selected item as the value. Then hide the results(dropdown)
+   *
+   * @param {Object} [result] when defined uses the result instead of the
+   * this.selected value
+   */
+  onSelect(result) {
+    if (result) {
+      this.selected = result;
+    }
+
+    this.value        = this.selected;
+    this.results = [];
+    this.justSelected = true;
+    this.search = this.label(this.value);
+    this.showResults  = false;
+  }
+
+  /**
+   * when search string changes perform a request, assign it to results
+   * and select the first result by default.
+   *
+   * @param {string} newValue
+   * @param {string} oldValue
+   *
+   * @returns {void}
+   */
+  searchChanged(newValue, oldValue) {
+    if (!this.shouldPerformRequest()) {
+      this.results = [];
+
+      return;
+    }
+
+    return this.findResults(this.searchQuery(this.search)).then(results => {
+      this.results = this.sort(results);
+
+      if (this.results.length !== 0) {
+        this.selected    = this.results[0];
+        this.value       = this.selected;
+        this.showResults = true;
+      }
+    });
+  }
+
+  /**
+   * when the results change, add the event listeners to the li items.
+   * This has to be done because bootstrap will otherwise overrule the events
+   */
+  resultsChanged() {
+    this.removeEventListeners(this.liEventListeners);
+    this.liEventListeners = [];
+
+    this.queue.queueTask(() => {
+      this.element.querySelectorAll('li').forEach((li, index) => {
+        let clickEventFunction = () => {
+          this.onSelect(this.results[index])
+        };
+
+        li.addEventListener('click', clickEventFunction);
+
+        this.liEventListeners.push({
+          callback : clickEventFunction,
+          eventName: 'click',
```

Comment:
It would move over the options and enter would select the active option.
The GIF in the description shows the behavior in detail.